### ISSUE TITLE
Argo CD not syncing the desired manifest state with the live manifest state showing out of sync.

### ISSUE THREAD

**Issue by cherrymu** [argoproj/argo-cd#20826]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

I am using argocd to maintain our helm charts lifecycle.  I am using the helm charts for application coming from this [repo](https://github.com/SeldonIO/helm-charts/tree/master) 

**To Reproduce**

1. Install the setup chart v2.8.3 from [here](https://github.com/SeldonIO/helm-charts/tree/master/helm-charts/seldon-core-v2-setup) using argo app
2. Upgrade the chart to v2.8.5
3. After that you will that the application health is healthy but only one of the CR resource named SeldonConfig will be out of sync showing difference as below between live and desired. 
4. Use the command kubectl get apps and it will show was OutOfSync and healthy 

**Expected behavior**
Live manifest should match with the desired manifest and states should be synced. 

**Screenshots**

[
![Screenshot from 2024-11-18 14-10-26](https://github.com/user-attachments/assets/846413f6-6510-44c2-b428-d5813b7013e8)
](url)
**Version**

```
v2.13.0
```

**Logs**
[logs_argocd_20241118_143823.tar.gz](https://github.com/user-attachments/files/17801971/logs_argocd_20241118_143823.tar.gz)

**Comment by andrii-korotkov-verkada** [comment]
I wonder if https://github.com/argoproj/gitops-engine/pull/640 would fix this too.

**Comment by cherrymu** [comment]
Thanks @andrii-korotkov-verkada, is there anyway to try the fix you have done to validate it with my issue?

**Comment by andrii-korotkov-verkada** [comment]
Yes, you can use replace in go.mod for gitops-engine to point at my fork + branch and run `go mod tidy`. Then you can build a custom image by using `make image`.

**Comment by andrii-korotkov-verkada** [comment]
You can also try playing around with https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/#known-kubernetes-types-in-crds-resource-limits-volume-mounts-etc.

**Comment by andrii-korotkov-verkada** [comment]
There's been updates in how the diff is calculated, available in v3 rc. Please, lmk if the issue reproduces there.

**Comment by neil-119** [comment]
I'm still seeing this as well. I have a bunch of apps (in app of apps pattern) and some of them are showing healthy but out of sync, but no matter what I do, it does not sync. I tried refresh, hard refresh, full sync, partial sync, etc. but it doesn't sync, no errors in resource events, nothing.
